pwa/src/templates/templateParts/mappingForm/MappingFormTemplate.tsx

```diff
@@ -80,19 +104,97 @@ export const MappingFormTemplate: React.FC<MappingFormTemplateProps> = ({ mappin
     basicFields.forEach((field) => setValue(field, mapping[field]));
 
     const newMapping = Object.entries(mapping.mapping).map(([key, value]) => ({ key, value }));
-    setMapping(newMapping);
+    setMappingKeyValue(newMapping);
+    setMappingCodeEditor(JSON.stringify(mapping.mapping.length ? mapping.mapping : {}, null, 2));
```

Comment:
I'm thinking we could maybe get rid of the `mappingCodeEditor`, `unsetCodeEditor`, ... altogether by making one generic function that simply returns the stringified value of the mapping.

E.g. (dummy code):

```
const getStringifiedCode = (type: "mapping" | "unset" | "cast") => { // duplicating the type here; we should make an actual Type
   
  return JSON.stringify(mapping[type].length ? mapping[type] : {}, null, 2);  

}
```

Now `getStringifiedCode("cast")` would hold the value of `castCodeEditor`, removing the need to keep track of the state (and requiring to JSON.stringify every time we need the value.